src/ratelimit/RateLimitController.test.ts

```diff
@@ -0,0 +1,132 @@
+import { ControllerMessenger } from '../ControllerMessenger';
+import {
+  ControllerActions,
+  RateLimitStateChange,
+  RateLimitController,
+  RateLimitMessenger,
+  GetRateLimitState,
+  CallApi,
+} from './RateLimitController';
+
+const name = 'RateLimitController';
+
+const implementations = { showNativeNotification: jest.fn() };
+
+type RateLimitedApis = typeof implementations;
+
+/**
+ * Constructs a unrestricted controller messenger.
+ *
+ * @returns A unrestricted controller messenger.
+ */
+function getUnrestrictedMessenger() {
+  return new ControllerMessenger<
+    GetRateLimitState<RateLimitedApis> | CallApi<RateLimitedApis>,
+    RateLimitStateChange<RateLimitedApis>
+  >();
+}
+
+/**
+ * Constructs a restricted controller messenger.
+ *
+ * @param controllerMessenger - An optional unrestricted messenger
+ * @returns A restricted controller messenger.
+ */
+function getRestrictedMessenger(
+  controllerMessenger = getUnrestrictedMessenger(),
+) {
+  return controllerMessenger.getRestricted<
+    typeof name,
+    ControllerActions<RateLimitedApis>['type'],
+    never
+  >({
+    name,
+    allowedActions: ['RateLimitController:call'],
+  }) as RateLimitMessenger<RateLimitedApis>;
+}
+
+const origin = 'snap_test';
+const message = 'foo';
+const successResult = { isRateLimited: false, result: undefined };
+
+describe('RateLimitController', () => {
+  jest.useFakeTimers();
+
+  it('action: RateLimitController:call', async () => {
+    const unrestricted = getUnrestrictedMessenger();
+    const messenger = getRestrictedMessenger(unrestricted);
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
Why set this variable then?

Reply:
It just looked weird to me to initialize the class and throw it away 😅 But can remove 👍

Reply:
Done!